strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/proxied/ProxyRepositoryArtifactResolver.java

```diff
@@ -67,14 +49,6 @@ public RepositoryPath fetchRemoteResource(RepositoryPath repositoryPath)
         throws IOException
     {
         Repository repository = repositoryPath.getFileSystem().getRepository();
-        logger.debug(String.format("Checking in [%s]...", repositoryPath));
-
-        final RepositoryPath candidate = preProxyRepositoryAccessAttempt(repositoryPath);
```

Comment:
I don't think we should remove this logic, we should check expiration in `preProxyRepositoryAccessAttempt` instead.

Reply:
Method `ProxyRepositoryArtifactResolver#fetchRemoteResource` should force fetching artifact from remote as its name states.

This method is called from 

- `ProxyRepositoryProvider#resolvePathForceFetch` (method name indicates `force` indeed)

 and 

- `MavenArtifactFetchedFromRemoteEventListener#downloadArtifactMetadataAtArtifactIdLevelFromRemote` (the method name is also meaningful)

If you want to resolve path (and check if the path exists locally in the meantime), you should use `ProxyRepositoryProvider#fetchPath`

Reply:
if you think that `ProxyRepositoryProvider#resolvePathForceFetch` then let's think about better name. But I think we shouldn't change related logic to suit your expectations regarding method name. :)
this and below logic needed to prevent race conditions within concurrent threads, there was an issue about this and currently it works fine. So we really shouldn't drop this

Reply:
let's try to clarify it from another side
there was an issue [SB-1133](https://youtrack.carlspring.org/issue/SB-1133)
so you want to say that your changes will not get it back?